appointment/upload-documents/uploadDocumentsController.js

```diff
@@ -96,6 +126,11 @@ define('uploadDocumentsController', [], function() {
 						$scope.isResidentialAppointment = response.residentialAppointment;
 						$scope.appointmentTimeStr = response.appointmentTimeStr;
 						$scope.uploadDeadlineStr = response.uploadDeadlineStr.toUpperCase();
+						$scope.appointmentId = response.appointmentId;
+						
+						// Checking to see if they've already filled out consent
+						const virtualAppointmentConsentId = response.consentId;
+						$scope.validateConsent(virtualAppointmentConsentId);
```

Comment:
Hm, I would probably send a value from the BE that says if they've previously consented or not--that way the BE can actually check the values, and we provide a stronger guarantee to the FE that there's actually been consent

Reply:
Hold up--what's the flow here? My expectation would be 1) Go to upload documents page, 2) validate my information, 3) If I haven't consented previously, I'm presented with consent page, 4) Can actually upload documents. 

That's what's happening right? I would still include in the BE response a boolean indicating if they've consented previously or not rather than checking this stuff on the FE / initiating another request to the BE

Reply:
Yes, that is the flow. When the data is validated, getClientInformation now also gets the virtualConsentId from the appointment table. If the ID is null, that means they haven't yet consented, and they are presented with the consent page. If there is an ID there, it accepts that as evidence that they've consented, since they aren't able to submit the consent form unless they've consented (said yes to virtual preparation and added some kind of characters to the signature field).

Do you think I should add a "valid" boolean to the consent table? Then before inputting values into the table, check to make sure the consent to virtual prep is true and the signature has some kind of string in it; if both those conditions are true, make the valid boolean true, otherwise make it false?

As it stands, this would be somewhat redundant because the only way their consent is added to the DB is if the consent fields are valid in the first place. Let me know if I communicated this well enough in my comments before and here, and if you think this is the way to go.

Reply:
I wouldn't add a valid boolean to the consent table--I would sooner pull the values from the table, and check them in PHP, and then include a `consented` boolean in the response from `getClientInformation`. Ik you say "they can't submit form if it's not valid"--but someone could just CURL the endpoint with invalid data and bypass this check the way it stands. I would check the information on the BE to avoid this situation.

Reply:
I see, thanks. Made that change.